jep/0000/README.adoc

```diff
@@ -0,0 +1,124 @@
+
+Both Jenkins core and plugins use cryptography that is not allowed and/or are not from an approved implementation, and this prevents adoption of Jenkins by these end users.
+Examples would be during setup where the install wizard creates a first admin user using the internal security realm which uses ``bcrypt`` which would fall fowl of the standard due to its use of ``blowfish`` which is not an approved algorithm.
+
+== Reasoning
+
+In order to be FIPS-140 compliant the encryption algorithms used need to be validated, this requires complex configuration of the host, the JVM as well as the changes to Jenkins and plugin code as well as checking of any plugin code that would be installed.  As such it is not considered that the Jenkins community supports, or contains instructions on how to run Jenkins in a compliant manner, but merely contains only an overview to say it may be possible with various configuration of the JVM, Servlet environment, limitations of plugins and finally instructing Jenkins to prefer a FIPS-140 mode of operation.
+
+As of the time of writing the validated Java implementations are only FIPS-140-2 compliant even though FIPS-140-3 has superseded it.  As such the initial implementations will be targeting the older FIPS-140-2 standard.
+
+== Backwards Compatibility
+
+It is out of scope to support previous configuration, as it would not be possible to migrate non FIPS-140 compliant data to FIPS-140 compliant data within a FIPS-140 compliant environment.  As such there is no backwards compatibility with previous versions.
+
+It may but is not a requirement of this JEP be possible to switch from a FIPS-140 compliant environment to a non FIPS-140 compliant environment.
+
+Code changes will however be backwards compatible; if the system is running in a non FIPS-140 compliant environment then it will continue to run and operate the same after any changes due to this JEP.
+
+== Security
+
+Some of the cryptography supported in FIPS 140 (especially FIPS 140-2) is considered less secure than some modern alternatives.  There is no alternative in cases like these where FIPS-140 compliance is required.
+
+== Infrastructure Requirements
+
+There are no new infrastructure requirements related to this proposal.
+
+== Testing
+
+New code that changes behaviour when it detects the desire to run in FIPS-140 mode would be covered by unit tests like any other code change.  There would be no new testing introduced in a FIPS-140 compliant environment due to the complexity and code of maintaining this setup.
+
+It is possible that regressions could be made due to the lack of automation to catch issues, however automation can only catch pre-known issues, and could not detected for example the introduction of a new encryption routine from a previously unknown library.
```

Comment:
This seems like a general statement of the "imperfect but useful" state of test automation.  I'd remove it (first preference) or rephrase it significantly (suggestion below).
```suggestion
Test automation is not expected to detect code in Jenkins core or in Jenkins plugins that is not FIPS-140 compliant.
```